Makefile

```diff
@@ -263,18 +271,50 @@ include mk/external.mk
 
 # Non-trivial demonstration programs
 ifeq ($(call has, SDL), 1)
-doom: $(BIN) $(DOOM_DATA)
-	(cd $(OUT); ../$(BIN) doom.elf)
+WEB_RESOURCES := web-resources
+WEB_JS_RESOURCES := $(WEB_RESOURCES)/js
+doom_action := (cd $(OUT); ../$(BIN) doom.elf)
+ifeq ("$(CC_IS_EMCC)", "1")
```

Comment:
The file `mk/external.mk` was meant to be related to retrieve something from external sources. Don't specify emscripten rules here.

Reply:
> The file `mk/external.mk` was meant to be related to retrieve something from external sources. Don't specify emscripten rules here.

I want to distinguish between the steps that need to be taken when utilizing emcc. Furthermore, when emcc is not being utilized, we do not want the emcc CFLAGS to be used. Notice that just emcc needs the TIMIDITY_DATA.

Reply:
Minimize the necessary changes when possible.

Reply:
I would like to embed all build executable and configuration files such as doomrc into wasm. Although the size getting larger, but we can select and run them in a single web page by selecting from a menu bar.

Reply:
> I would like to embed all build executable and configuration files such as doomrc into wasm.

It sounds like a good idea.

Reply:
> I would like to embed all build executable and configuration files such as doomrc into wasm. Although the size getting larger (fetching time might be longer but only once because it will be cached by the browser for further fetch), but we can select and run them in a single web page from a menu bar.
> 
> Consequently, we do not need to check the makefile build target and embed corresponding files respectively.

The final wasm is 61.40 MB,  this is larger than GitHub's recommended maximum file size of 50.00 MB. I am going to test it out if this is fine to host on GitHub Pages.

Reply:
> > I would like to embed all build executable and configuration files such as doomrc into wasm. Although the size getting larger (fetching time might be longer but only once because it will be cached by the browser for further fetch), but we can select and run them in a single web page from a menu bar.
> > Consequently, we do not need to check the makefile build target and embed corresponding files respectively.
> 
> The final wasm is 61.40 MB, this is larger than GitHub's recommended maximum file size of 50.00 MB. I am going to test it out if this is fine to host on GitHub Pages.

Github Pages are fine even exceeds GitHub's recommended maximum file size of 50.00 MB.